### PR TITLE
Initial commit for is_sorted method on Field

### PR DESCRIPTION
**Pull request by atbenmurray**
Work still to do:
 * Tests for is_sorted
 * is_sorted added to abstract Field once that it goes through

### CODE REVIEW

File: exetera/core/dataframe.py [hunk 1]
```diff
@@ -29,15 +34,20 @@ def __init__(self, name, dataset,data=None,h5group:h5py.Group=None):
...
-        if isinstance(dataset,dst.HDF5Dataset):
-            dataset[name]=self
-        if data is not None:
-            if isinstance(data,dict) and isinstance(list(data.items())[0][0],str) and isinstance(list(data.items())[0][1], fld.Field) :
-                self.fields=data
-        elif h5group is not None and isinstance(h5group,h5py.Group):
+        if isinstance(dataset, dst.HDF5Dataset):
+            dataset[name] = self
+        if dataframe is not None:
+            if isinstance(dataframe, dict) and isinstance(list(dataframe.items())[0][0], str) and\
```
Comment (atbenmurray) [inline exetera/core/dataframe.py:44]: Validate all entries, not just the first one

File: exetera/core/dataframe.py [hunk 2]
```diff
@@ -29,15 +34,20 @@ def __init__(self, name, dataset,data=None,h5group:h5py.Group=None):
...
+                    isinstance(list(dataframe.items())[0][1], fld.Field):
+                self.fields = dataframe
+            else:
+                raise ValueError("if set, 'dataframe' must be a dictionary mapping strings to fields")
+
+        elif h5group is not None and isinstance(h5group, h5py.Group):
             fieldtype_map = {
```
Comment (atbenmurray) [inline exetera/core/dataframe.py:51]: The newly constructed dataframe has API for adding fields, so maybe we can use that

File: exetera/core/dataset.py [hunk 3]
```diff
@@ -9,76 +9,87 @@
...
-
-    def __iter__(self):
-        pass
-
-    def __next__(self):
-        pass
-
-    def __len__(self):
-        pass
+# class Dataset():
```
Comment (atbenmurray) [inline exetera/core/dataset.py:12]: Remove dead code

File: exetera/core/session.py [hunk 4]
```diff
@@ -652,42 +652,82 @@ def create_like(self, field, dest_group, dest_name, timestamp=None, chunksize=No
...
+                raise ValueError("'group' must be an Exetera DataFrame but a "
+                                 "{} was passed to it".format(type(group)))
+
+        if isinstance(group, h5py.Group):
+            return fld.categorical_field_constructor(self, group, name, timestamp, chunksize)
+        else:
+            return group.create_timestamp(name, timestamp, chunksize)
 
 
+    #TODO: update this method for the existence of dataframes
```
Comment (atbenmurray) [inline exetera/core/session.py:730]: TODO is a no-op as this code works for both groups and dataframes; remove the comment

File: tests/test_dataset.py [hunk 5]
```diff
@@ -3,18 +3,19 @@
 from exetera.core import session
 from io import BytesIO
 
+
```
Comment (atbenmurray) [inline tests/test_dataset.py:6]: These tests could do with augmenting for the other field types

**Review by deng113jie — APPROVED** [review]
Minor updates needed (issue #138 ), refer to the inline review for details.

**Pull request merged**